src/Router.cpp

```diff
@@ -110,50 +130,66 @@ void Router::splitTarget(const std::string &target)
 	}
 }
 
-bool Router::pathIsValid(HTTPRequest &request, std::string webRoot)
+bool Router::pathIsValid(HTTPRequest &request, std::string &webRoot)
 {
 	std::string host = request.getHost();
+	std::cout << "pathIsValid Host: " << host << std::endl;
+	std::cout << "pathIsValid WebrRoot: " << webRoot << std::endl;
 	size_t pos = host.find(":");
 	if (pos != std::string::npos)
 	{
 		host = host.substr(0, pos);
 	}
-	std::string pathWithQuery = request.getRequestTarget();
-	std::string path = pathWithQuery.substr(0, pathWithQuery.find("?"));
+	// std::cout << "Host (after : trailing) :" << host << std::endl;
 
-	if (host == "localhost")
-		path = webRoot + path;
-	else
-		path = webRoot + "/" + host + path;
+	// for ease of use during deployment
+	// this if/else allows to reach target with tester or browser
+	// if (host != "localhost")
+	// 	webRoot = host + webRoot;
 
-	std::cout << std::endl << "Path: " << path << std::endl << std::endl;
-	std::cout << "Path: " << path << std::endl;
+	std::cout << std::endl << "pathIsValid Path: " << webRoot << std::endl << std::endl;
+	std::cout << "pathIsValid Path: " << webRoot << std::endl;
 	struct stat buffer;
-	if (stat(path.c_str(), &buffer) != 0)
+	if (stat(webRoot.c_str(), &buffer) != 0)
 	{
+		std::cout << "+-+-+-+-+- does not exist" << std::endl;
+		request.setPath(webRoot);
 		return false;
 	}
 	if (S_ISDIR(buffer.st_mode))
 	{
 		// std::cout << "Path is a directory" << std::endl;
-		if (!path.empty() && path[path.length() - 1] != '/')
+		if (!webRoot.empty() && webRoot[webRoot.length() - 1] != '/')
+		{
+			webRoot += "/";
+		}
+		// IF THIS IS NOT DEFINE IN THE CONFIG FILE, THEN DO WHAT IS BELOW:
+		if (_serverBlock.getIndex().empty())
+		{
+			webRoot += "index.html";
+		}
+		else
 		{
-			path += "/";
+			std::cout << "Index: " << _serverBlock.getIndex()[0] << std::endl;
+			std::string index = _serverBlock.getIndex()[0];
+			webRoot += index;
+			std::cout << "Path: " << webRoot << std::endl;
 		}
```

Comment:
You are taking here the first value of getIndex, but we should check for all of them, right? This is a real question cause I don't know exactly how this work, should work.

Reply:
Yes that will be in next PR, if the first index is not available, I should check the second, etc...